checker/checker.go

```diff
@@ -258,21 +267,27 @@ func (c *Checker) Close() {
 		return
 	}
 
+	c.closeDBs()
+
+	c.closed.Set(true)
+}
+
+func (c *Checker) closeDBs() {
 	for _, instance := range c.instances {
 		if instance.sourceDB != nil {
 			if err := dbutil.CloseDB(instance.sourceDB); err != nil {
 				log.Errorf("close source db %+v error %v", instance.sourceDBinfo, err)
 			}
+			instance.sourceDB = nil
```

Comment:
is it necessary to set to `nil`? the `subtask` will be always recycled if `Init` failed

Reply:
not necessary. only want to do the pattern (release then reset) and not rely on other code.